Core/HLE/sceNetAdhoc.cpp

```diff
@@ -113,8 +120,457 @@ void __NetAdhocShutdown() {
 	}
 }
 
+static void __AdhocctlNotify(u64 userdata, int cyclesLate) {
+	SceUID threadID = userdata >> 32;
+	int uid = (int)(userdata & 0xFFFFFFFF);
+
+	s64 result = 0;
+	u32 error = 0;
+
+	SceUID waitID = __KernelGetWaitID(threadID, WAITTYPE_NET, error);
+	if (waitID == 0 || error != 0)
+		return; // FIXME: Is it safe to exit here like this without re-scheduling the event? Will this event be triggered again? What will happen to the result i might want to change if exited here?
+
+	int waitVal = __KernelGetWaitValue(threadID, error); // FIXME: Is this value going to be a valid value if waitID == 0? or it's a value belonged to other event?
```

Comment:
If you got an invalid wait ID, it could be this thread could even be waiting on something totally different now.  So no, the waitVal is probably not interesting in that case.

-[Unknown]